dataset/include/scipp/dataset/dataset.h

```diff
@@ -11,6 +11,7 @@
 #include <unordered_map>
 
 #include <boost/iterator/transform_iterator.hpp>
+#include <boost/range/adaptor/map.hpp>
```

Comment:
Can we avoid such an include in a central header?

Reply:
Yes no longer needed now functions have been moved.